public/app/features/dashboard/components/PanelEditor/PanelEditorTabs.tsx

```diff
@@ -17,77 +17,65 @@ interface PanelEditorTabsProps {
   onChangeTab: (tab: PanelEditorTab) => void;
 }
 
-export class PanelEditorTabs extends PureComponent<PanelEditorTabsProps> {
-  private eventSubs = new Subscription();
+export const PanelEditorTabs: FC<PanelEditorTabsProps> = React.memo(({ panel, dashboard, tabs, onChangeTab }) => {
+  const forceUpdate = useForceUpdate();
```

Comment:
this seems like a pretty big code smell imho - i wonder what other bugs are lying around because state is mutated (?). Is it not feasable for the props to actually change when they... change?

(i saw a similar pattern elsewhere in the code - I don't think this should continue to spread)

Reply:
yes, agree 100% , this is due to some leftover panel model ways from angular that is mutable.